src/app/account/settings/ExportTransactions/index.tsx

```diff
@@ -0,0 +1,74 @@
+"use client"
+import React, { use, useEffect, useState } from 'react'
+import { LocalStorage, decryptTransactions, getTransactionsFromSupabase, getUserId } from '@/lib/utils';
+import { useSupabase } from '@/app/supabase-provider';
+import { SupabaseClient } from '@supabase/auth-helpers-nextjs';
+import { useToast } from '@/app/components/ui/use-toast';
+
+function objectArrayToCsvString(array:any[],delimiter:string=',',headers:any[]){
+  const rows = array.map(obj => [
+    obj.date,
+    obj.concept,
+    obj.amount,
+    obj.category
+  ]);
+  return [headers, ...rows].map(row => row.join(delimiter)).join('\n');
+};
+
+function downlaodCSV(data:any[]){
+  const delimeter = localStorage.getItem(LocalStorage.delimiter);
+  const csvString = objectArrayToCsvString(data,delimeter??",",['date', 'concept', 'amount', 'category'])
+
+  const blob = new Blob([csvString], { type: 'text/csv' });
+    const url = URL.createObjectURL(blob);
+
+    // Create a temporary link and trigger the download
+    const link = document.createElement('a');
+    link.href = url;
+    link.setAttribute('download', 'data.csv'); // Name of the file
+    document.body.appendChild(link);
+    link.click();
+    document.body.removeChild(link);
+
+}
+
+
+export function ExportTransactions() {
+  
+  const [isLoading,setIsLoading] = useState<boolean>(false);
+  const { supabase } = useSupabase();
+  const { toast } = useToast();
+
+  async function getAllTransactions(){
+    setIsLoading(true);
+    const data= await getTransactionsFromSupabase(supabase);
+    const userId = await getUserId(supabase);
+    if (!data) {
+      toast({
+        description:
+          "❎ Error fetching transactions. Please, try again later.",
+      });
+    } else {
+      const decryptData = decryptTransactions(data,userId?userId:"");
+      downlaodCSV(decryptData);
+    }
+    setIsLoading(false);
+  }
+  
+  return (
+    <div className='p-2 mt-2 pl-0'>
+      <span className='text-xl'> Would you like to export transactions in .csv</span>
```

Comment:
Can you please use the same classes than in other parts of Settings: 
<img width="550" alt="image" src="https://github.com/user-attachments/assets/af78ecf3-f7b4-44e3-a6a1-5aa394c04b79">
It doesn't match the same style of the whole Settings component: 
<img width="539" alt="image" src="https://github.com/user-attachments/assets/fe77e78f-0d72-4d24-83af-390cda5e1362">

Also, could you please change the texts? 
It would be great to keep the coherance. like: 
```
## Export (title)
Export all your transactions into a .csv file (description)
<Button>
```